### PR TITLE
Feat/task 4748

### PR DESCRIPTION
**Pull request by sureshkrishnan-mr**
[XXXX-<Title> - Please use the Work Item number and Title as PR Name, not subtasks]

#### 📲 What

Refactor multiple REST Controller to a single By-Resource REST Controller
#### 🤔 Why

Why it's needed, background context.

#### 🛠 How

More in-depth discussion of the change or implementation.

#### 👀 Evidence

Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR

#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/MenuController.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    Menu menu =
+        menuQueryService
+            .findById(id)
+            .orElseThrow(() -> new MenuNotFoundException(new MenuCommand(correlationId, id)));
+    return ResponseEntity.ok(menuMapper.toDto(menu));
+  }
+
+  @PutMapping(
+      value = "/{id}",
+      consumes = MediaType.APPLICATION_JSON_VALUE,
```
Comment (sdavis-amido) [inline java/src/main/java/com/amido/stacks/workloads/menu/api/v1/MenuController.java:175]: Do we actually need these produces and consumes for it to work?

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/CategoryController.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  @PostMapping
+  @Operation(
+      tags = "Category",
+      summary = "Create a category in the menu",
+      description = "Adds a category to menu",
+      operationId = "AddMenuCategory")
+  @CreateAPIResponses
+  ResponseEntity<ResourceCreatedResponse> addMenuCategory(
```
Comment (sdavis-amido) [inline java/src/main/java/com/amido/stacks/workloads/menu/api/v1/CategoryController.java:65]: Maybe just call this "addCategory"?

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/CategoryController.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  @PutMapping("/{categoryId}")
+  @Operation(
+      tags = "Category",
+      summary = "Update a category in the menu",
+      description = "Update a category to menu",
+      operationId = "UpdateMenuCategory")
+  @UpdateAPIResponses
+  ResponseEntity<ResourceUpdatedResponse> updateMenuCategory(
```
Comment (sdavis-amido) [inline java/src/main/java/com/amido/stacks/workloads/menu/api/v1/CategoryController.java:85]: Maybe just call this "updateCategory"?

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/ItemController.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  }
+
+  @PostMapping
+  @Operation(
+      tags = "Item",
+      summary = "Add an item to an existing category in a menu",
+      description = "Adds a menu item",
+      operationId = "AddMenuItem")
+  @CreateAPIResponses
+  ResponseEntity<ResourceCreatedResponse> addMenuItem(
```
Comment (sdavis-amido) [inline java/src/main/java/com/amido/stacks/workloads/menu/api/v1/ItemController.java:65]: Maybe just call this "addItem"?

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/CategoryController.java (hunk 2 above)
Comment (sureshkrishnan-mr) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v1/CategoryController.java]: done

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/CategoryController.java (hunk 3 above)
Comment (sureshkrishnan-mr) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v1/CategoryController.java]: done

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/ItemController.java (hunk 4 above)
Comment (sureshkrishnan-mr) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v1/ItemController.java]: done

File: java/src/main/java/com/amido/stacks/workloads/menu/api/v1/MenuController.java (hunk 1 above)
Comment (sureshkrishnan-mr) [reply java/src/main/java/com/amido/stacks/workloads/menu/api/v1/MenuController.java]: done - tested without produces & consumes and it worked fine.

**Pull request merged**